### ISSUE TITLE
SC2024 should not trigger on input redirection

### ISSUE THREAD

**Issue by jasonaowen** [koalaman/shellcheck#2070]
#### For bugs
- Rule Id (if any, e.g. SC1000): [SC2024](https://github.com/koalaman/shellcheck/wiki/SC2024)
- My shellcheck version (`shellcheck --version` or "online"): 0.4.6
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

# Preferred construction, fails with SC2024
sudo mysql < seed.sql

# Alternate construction, fails instead with SC2002
cat seed.sql | sudo mysql
```

#### Here's what shellcheck currently says:

```
Line 4:
sudo mysql < seed.sql
           ^-- SC2024: sudo doesn't affect redirects. Use sudo cat file | ..
 
Line 7:
cat seed.sql | sudo mysql
    ^-- SC2002: Useless cat. Consider 'cmd < file | ..' or 'cmd file | ..' instead.
```

#### Here's what I wanted or expected to see:

No SC2024 warning.

The rule seems to be solely concerned with writing to files as the current rather than the `sudo`'d user. Input redirection with `<` is unrelated to that concern.

The alternate construction of using `cat`, kind of how the inline warning message suggests, does make explicit what user is reading the input file, but instead causes a SC2002 useless cat warning.

**Comment by jabberabbe** [comment]
Apparently this is intended behavior. The case is specifically covered at line 1079 of `src/ShellCheck/Checks/Commands.hs`, and there is a specific test for input redirection at line 1062. Although it might seem reasonable if you're reading from a read-protected file, I agree that mostly input redirection just works because usually files are readable by normal users.
The wiki page reports only examples for `tee` but it is noted that as an exception you can safely ignore the warning if you want to redirect using the current user's permissions. You can use an ignore directive to disable this warning: https://github.com/koalaman/shellcheck/wiki/Ignore.
This issue can be safely closed.

**Comment by koalaman** [comment]
You're both right. The warning is indeed as intended, but the wiki is misleading in only talking about output. I've updated it to cover input as well. Thanks!
